subsys/shell/shell.c

```diff
@@ -786,18 +782,26 @@ static void shell_state_collect(const struct shell *shell)
 
```

Comment:
wouldn't it be possible to reduce number of history_exit calls in that case?
it should stay in history only in case of escape, \0 and metakey_handle. Maybe a flag history_exit (initialy true) and cleared in those particular cases and then `if(history_exit) history_mode_exit(shell)`. It will reduce code size with less function calls.

Reply:
good point, I will update that